src/phoenix/server/api/dataloaders/latency_ms_quantile.py

```diff
@@ -1,89 +1,206 @@
+) -> AsyncIterator[Tuple[ResultPosition, QuantileValue]]:
+    if dialect == POSTGRESQL:
+        results = _get_results_postgresql(session, segment, params)
+    elif dialect == SQLITE:
+        results = _get_results_sqlite(session, segment, params)
+    else:
+        assert_never(dialect)
+    async for position, quantile_value in results:
+        yield position, quantile_value
+
+
+async def _get_results_sqlite(
+    session: AsyncSession,
+    segment: Segment,
+    params: Mapping[Param, List[ResultPosition]],
+) -> AsyncIterator[Tuple[ResultPosition, QuantileValue]]:
+    kind, (start_time, end_time), filter_condition = segment
+    projects_per_prob: DefaultDict[Probability, List[ProjectRowId]] = defaultdict(list)
+    for project_rowid, probability in params.keys():
+        projects_per_prob[probability].append(project_rowid)
+    for probability, project_rowids in projects_per_prob.items():
+        if kind == "trace":
+            pctl = func.percentile(models.Trace.latency_ms, probability * 100)
+        elif kind == "span":
+            pctl = func.percentile(models.Span.latency_ms, probability * 100)
+        else:
+            assert_never(kind)
+        stmt = (
+            select(
+                models.Trace.project_rowid,
+                pctl,
+            )
+            .where(models.Trace.project_rowid.in_(project_rowids))
+            .group_by(models.Trace.project_rowid)
         )
-    if start_time:
-        return and_(
-            models.Project.id == id_,
-            start_time <= models.Trace.start_time,
+        if kind == "trace":
+            if start_time:
+                stmt = stmt.where(start_time <= models.Trace.start_time)
+            if end_time:
+                stmt = stmt.where(models.Trace.start_time < end_time)
+        elif kind == "span":
+            stmt = stmt.join(models.Span)
+            if start_time:
+                stmt = stmt.where(start_time <= models.Span.start_time)
+            if end_time:
+                stmt = stmt.where(models.Span.start_time < end_time)
+            if filter_condition:
+                span_filter = SpanFilter(condition=filter_condition)
+                stmt = span_filter(stmt)
+        else:
+            assert_never(kind)
+        data = await session.stream(stmt)
+        async for project_rowid, quantile_value in data:
+            for position in params[(project_rowid, probability)]:
+                yield position, quantile_value
+
+
+async def _get_results_postgresql(
+    session: AsyncSession,
+    segment: Segment,
+    params: Mapping[Param, List[ResultPosition]],
+) -> AsyncIterator[Tuple[ResultPosition, QuantileValue]]:
+    kind, (start_time, end_time), filter_condition = segment
+    probs_per_project: DefaultDict[ProjectRowId, List[Probability]] = defaultdict(list)
+    for project_rowid, probability in params.keys():
+        probs_per_project[project_rowid].append(probability)
+    pp: Values = values(
```

Comment:
Not sure I know what this var name stands for. Abbreviations for the columns?

Reply:
I think another one where it would make sense to have separate functions for trace vs. span.

Reply:
> Not sure I know what this var name stands for.

`Values` is a temporary table, and its name is specified below as `project_probabilities`. `pp` is its alias, or in fancier terms, its _nom de guerre_.